### PR TITLE
expose json configuration setting and use them in both serializers

### LINKED ISSUE #10: Serialization is happening twice for each component with different settings. Setting are not exposed.
So far current flow is implemented:

1. Props are send to external process to render using `Jering.Javascript.NodeJS`
2. `Jering.Javascript.NodeJS` serialize object with `IJsonService `and it's default implementation. 
3. We return output html to cshtml
4. Then we serialize again props using different serialization options as in the line:

https://github.com/fortedigital/Forte.React.AspNetCore/blob/34ab5a3ed9bdd5e51b94abcfda905d3bab6e3bb6/Forte.React.AspNetCore/React/ReactService.cs#L105

Issues:
Serialization options are not shared, we can have different object on rendering html and hydrate.

### PR DESCRIPTION
**Pull request by pawil31**
PR fixing issue #10 
Json Serialization settings are exposed and reused during serialization.

### CODE REVIEW

File: Forte.React.AspNetCore/Configuration/ReactJsonSerializerOptions.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Text.Json;
+
+namespace Forte.React.AspNetCore.Configuration;
+
+public class ReactJsonSerializerOptions
+{
+    public JsonSerializerOptions JsonSerializerOptions { get; } = new()
```
Comment (konrad-galczynski) [inline Forte.React.AspNetCore/Configuration/ReactJsonSerializerOptions.cs]: please consider renaming to Options as context is provided by class name

File: Forte.React.AspNetCore/JsonSerializationService.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    private readonly JsonSerializerOptions _jsonSerializerOptions;
+
+    public JsonSerializationService(JsonSerializerOptions jsonSerializerOptions)
+    {
+        _jsonSerializerOptions = jsonSerializerOptions;
+    }
+
+    public ValueTask<T?> DeserializeAsync<T>(Stream stream,
+        CancellationToken cancellationToken = new CancellationToken())
```
Comment (konrad-galczynski) [inline Forte.React.AspNetCore/JsonSerializationService.cs]: as cancellation token is not used please consider using CancellationToken.None

File: Forte.React.AspNetCore/JsonSerializationService.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        => JsonSerializer.DeserializeAsync<T>(stream, _jsonSerializerOptions, cancellationToken);
+
+    public Task SerializeAsync<T>(Stream stream, T value, CancellationToken cancellationToken = new CancellationToken())
```
Comment (konrad-galczynski) [inline Forte.React.AspNetCore/JsonSerializationService.cs]: as cancellation token is not used please consider using CancellationToken.None

File: Forte.React.AspNetCore/Configuration/ReactJsonSerializerOptions.cs (hunk 1 above)
Comment (pawil31) [reply Forte.React.AspNetCore/Configuration/ReactJsonSerializerOptions.cs]: Fixed

File: Forte.React.AspNetCore/JsonSerializationService.cs (hunk 2 above)
Comment (pawil31) [reply Forte.React.AspNetCore/JsonSerializationService.cs]: Fixed

File: Forte.React.AspNetCore/JsonSerializationService.cs (hunk 3 above)
Comment (pawil31) [reply Forte.React.AspNetCore/JsonSerializationService.cs]: Fixed

**Review by konrad-galczynski — APPROVED** [review]

**Review by wredzio — APPROVED** [review]

**Pull request merged**